sleap_roots/graphpipeline.py

```diff
@@ -382,79 +386,74 @@ def get_traits_value_plant_summary(
-                data_plant_frame_summary.at[
-                    j, NON_SCALAR_TRAITS[i] + "_fprc75"
-                ] = np.nan
-                data_plant_frame_summary.at[
-                    j, NON_SCALAR_TRAITS[i] + "_fprc95"
-                ] = np.nan
+        trait = data_plant_df[NON_SCALAR_TRAITS[i]]
+
+        data_plant_frame_summary_non_scalar[
+            NON_SCALAR_TRAITS[i] + "_fmin"
+        ] = trait.apply(lambda x: np.nanmin(x) if (len(x) > 0) else np.nan)
+        data_plant_frame_summary_non_scalar[
+            NON_SCALAR_TRAITS[i] + "_fmax"
+        ] = trait.apply(lambda x: np.nanmax(x) if len(x) > 0 else np.nan)
+        data_plant_frame_summary_non_scalar[
+            NON_SCALAR_TRAITS[i] + "_fmean"
+        ] = trait.apply(lambda x: np.nanmean(x) if len(x) > 0 else np.nan)
+        data_plant_frame_summary_non_scalar[
+            NON_SCALAR_TRAITS[i] + "_fmedian"
+        ] = trait.apply(lambda x: np.nanmedian(x) if len(x) > 0 else np.nan)
+        data_plant_frame_summary_non_scalar[
+            NON_SCALAR_TRAITS[i] + "_fstd"
+        ] = trait.apply(lambda x: np.nanstd(x) if len(x) > 0 else np.nan)
+        data_plant_frame_summary_non_scalar[
+            NON_SCALAR_TRAITS[i] + "_fprc5"
+        ] = trait.apply(
+            lambda x: np.percentile(x[~pd.isna(x)], 5)
+            if len(x[~pd.isna(x)]) > 0
+            else np.nan
+        )
+        data_plant_frame_summary_non_scalar[
+            NON_SCALAR_TRAITS[i] + "_fprc25"
+        ] = trait.apply(
+            lambda x: np.percentile(x[~pd.isna(x)], 25)
+            if len(x[~pd.isna(x)]) > 0
+            else np.nan
+        )
+        data_plant_frame_summary_non_scalar[
+            NON_SCALAR_TRAITS[i] + "_fprc75"
+        ] = trait.apply(
+            lambda x: np.percentile(x[~pd.isna(x)], 75)
+            if len(x[~pd.isna(x)]) > 0
+            else np.nan
+        )
+        data_plant_frame_summary_non_scalar[
+            NON_SCALAR_TRAITS[i] + "_fprc95"
+        ] = trait.apply(
+            lambda x: np.percentile(x[~pd.isna(x)], 95)
+            if len(x[~pd.isna(x)]) > 0
+            else np.nan
+        )
 
     # get summarized scalar traits per plant
-    data_plant_summary = pd.DataFrame([{"plant_name": data_plant["plant_name"][0]}])
-    column_names = data_plant.columns.tolist()
+    column_names = data_plant_df.columns.tolist()
+    data_plant_frame_summary = {}
     for i in range(len(SCALAR_TRAITS)):
         if SCALAR_TRAITS[i] in column_names:
-            trait = data_plant[SCALAR_TRAITS[i]].values
+            trait = data_plant_df[SCALAR_TRAITS[i]]
+            # LW start
+            if trait.shape[0] > 0:
+                if not (
+                    type(trait[0]) == int
+                    or type(trait[0]) == np.int64
+                    or type(trait[0]) == np.float64
+                    or type(trait[0]) == float
+                ):
+                    values = np.array([element[0] for element in trait])
```

Comment:
Won't this break if `trait` contains scalars? Like if `type(trait[0]) == int`, then for example:

```
trait = [0, 1, 2]
assert type(trait[0]) == int
element = trait[0]
element[0]  # error
```

Reply:
I remembered that the trait is like: [[0,1,2]].
The test of the pipeline is good, but I met a problem using Spyder to run code with new environment, will fix it soon

Reply:
I will modify it together with issue #33